autocertkit/utils.py

```diff
@@ -1166,74 +1166,123 @@ def get_context_test_ifs(vm_ref):
     return [test_if for test_if in get_context_vm_ifs(vm_ref) if test_if[0] != mdevice]
 
 
-def wait_for_ip(session, vm_ref, device, timeout=300):
-    """Wait for an IP address to be returned (until a given timeout)"""
+def get_management_vif(session, vm_ref):
+    networks = get_management_network(session)
+    for vif in session.xenapi.VM.get_VIFs(vm_ref):
+        if session.xenapi.VIF.get_network(vif) in networks:
+            return vif
+    return None
 
-    # Check to see if a static IP has been configured
-    xs_data = session.xenapi.VM.get_xenstore_data(vm_ref)
-    log.debug("xenstore data: %s" % xs_data)
-    key = 'vm-data/control/%s/ip' % device
-    if key in xs_data.keys():
-        log.debug("Static IP %s found for VM %s" % (xs_data[key], vm_ref))
-        return xs_data[key]
 
-    if session.xenapi.VM.get_is_control_domain(vm_ref):
-        ipaddr = _get_control_domain_ip(session, vm_ref, device)
-        log.debug("Control domain %s has IP %s on device %s" %
-                  (vm_ref, ipaddr, device))
-        return ipaddr
+def init_vms_first_run(session, vm_refs):
+    for vm_ref in vm_refs:
+        mip = get_context_vm_mip(vm_ref)
+        # Install SSH Keys for Plugin operations
+        call_ack_plugin(session, 'inject_ssh_key',
+                        {'vm_ref': vm_ref, 'mip': mip, 'username': 'root',
+                         'password': DEFAULT_PASSWORD})
+        # Ensure that we make sure the switch accesses IP addresses by
+        # their own interfaces (avoid interface forwarding).
+        call_ack_plugin(session, 'reset_arp', {'vm_ref': vm_ref, 'mip': mip})
 
-    if not device.startswith('eth'):
-        raise Exception(
-            "Invalid device specified, it should be in the format 'ethX'")
 
-    start = time.time()
+def disable_vm_static_ip_service(session, mip):
+    cmd = """systemctl stop static-ip; systemctl disable static-ip; systemctl status static-ip; """ \
+          """rm -f /etc/sysconfig/network-scripts/ifcfg-eth*; """
+    ssh_command(mip, 'root', DEFAULT_PASSWORD, cmd, attempts=1, timeout=60)
 
-    def should_timeout(start, timeout):
-        """Method for evaluating whether a time limit has been met"""
-        return time.time() - start > float(timeout)
 
-    i = 0
-    while not should_timeout(start, timeout):
-        log.debug("Trying to retrieve VM IP address - Attempt %d" % i)
-        ips = get_vm_ips(session, vm_ref)
-        log.debug("VM %s has these %s IPs" % (vm_ref, ips))
+def get_vm_vif_ifs(session, vm_ref):
+    """Get vm interfaces information (vif, mac, ip) from xenstore vif records"""
+    ifs = {}
+    dom_root = "/local/domain/%s" % str(session.xenapi.VM.get_domid(vm_ref))
 
-        for k, v in ips.iteritems():
-            if k == device:
-                return v
+    cmd = b'''xenstore-ls -f %s | grep -E "%s/device/vif/[0-9]+/mac|%s/attr/vif/[0-9]+/ipv4/[0-9]+"''' % (dom_root, dom_root, dom_root)
```

Comment:
Using `str.format()` can aviod the redundant `dom_root` paramaters.

Reply:
I won't break the style, you can see in all sources, format() is not used.